lib/liquid/standardfilters.rb

```diff
@@ -869,6 +869,31 @@ def default(input, default_value = '', options = {})
       false_check || (input.respond_to?(:empty?) && input.empty?) ? default_value : input
     end
 
+    # @liquid_public_docs
+    # @liquid_type filter
+    # @liquid_category array
+    # @liquid_summary
+    #   Returns the sum of all elements in an array.
+    # @liquid_syntax array | sum
+    # @liquid_return [number]
+    def sum(input, property = nil)
+      ary = InputIterator.new(input, context)
+
+      return 0 if ary.empty?
+
+      if property.nil?
+        # If we're doing an ordinary sum, all values must be numeric
+        raise(Liquid::ArgumentError, "cannot sum non-numeric values") unless ary.all? { |el| el.is_a?(Numeric) }
```

Comment:
This should probably use `Utils.to_number` for consistency with the `plus` filter.

Reply:
Is your suggestion `...unless ary.all? { |el| Utils.to_number(el).is_a?(Numeric) }`

In other words, not just checking if the element is a Number, but if it can be coerced into to a number? (e.g., numeric strings like "123")

Reply:
Looking at `to_number`, it seems that for anything that is not a `Float`, `Numeric`, or `String` and does not respond to `:to_number`, it returns `0`.

Would we want `{{ [1, 2, 3, "a"] | sum }}` to return `6` or raise a `Liquid::ArgumentError`?